recipe/meta.yaml

```diff
@@ -16,13 +16,14 @@ build:
 
 requirements:
   host:
-    - python >=3.11
+    - python >=3.10
     - wheel
     - setuptools >=64
     - setuptools-scm >=8
     - pip
   run:
-    - python >=3.11
+    - typing-extensions
```

Comment:
I think this is needed only if `python < 3.12`

Reply:
In conda you can't have dependencies just in specific python versions for `noarch: python` packages.

Reply:
Ow so you're saying that there is nothing similar to `# [not win]` modifiers?

Reply:
There is, but those consider the platform and/or the dependency version used when the package is built by `conda-build` (in this case `python==3.12`). So if you add a pre-processor modifier that drop the `typing-extensions` package if `python == 3.12`, the dependency will be removed by the generated package, but given that for `noarch: python` you generate a single package for all python versions, when this package will be installed in python 3.10, it will miss the required `typing-extensions` dependency.

Reply:
The right name is _selector_, and for the records, here below the documentation:

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors

Reply:
Thanks for the explanation @traversaro, now it's more clear.

Reply:
See also https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python